paddle/fluid/operators/cross_entropy_op.cc

```diff
@@ -32,14 +32,18 @@ class CrossEntropyOp : public framework::OperatorWithKernel {
     int rank = x_dims.size();
     PADDLE_ENFORCE_EQ(rank, label_dims.size(),
                       "Input(X) and Input(Label) shall have the same rank.");
-    PADDLE_ENFORCE_EQ(framework::slice_ddim(x_dims, 0, rank - 1),
-                      framework::slice_ddim(label_dims, 0, rank - 1),
-                      "Input(X) and Input(Label) shall have the same shape "
-                      "except the last dimension.");
+    if (ctx->IsRuntime()) {
+      PADDLE_ENFORCE_EQ(framework::slice_ddim(x_dims, 0, rank - 1),
+                        framework::slice_ddim(label_dims, 0, rank - 1),
+                        "Input(X) and Input(Label) shall have the same shape "
+                        "except the last dimension.");
+    }
```

Comment:
Why not compare each positive dim and skip all negative dims in compile time?

Reply:
@sneaxiy Thanks! Add check condition, only skip checking when un-know the x and label dimensions in compile time :

```c++
 if ((!ctx->IsRuntime()) && (framework::product(x_dims) <= 0 ||
      framework::product(label_dims) <= 0)) {

}
```